### PR TITLE
Ref issue 10 wrapper to build a SOLVCON cluster.

### PR DESCRIPTION
**Pull request by tai271828**
This is the prototype of the project 1 https://github.com/solvcon/solvcon-gce/projects/1 . This commit has many hard-coded lines and bad argument design. However it works, so I think it would be good to create a PR for the team to have a discussion thread. Comments are very welcome and appreciated, so I could improve this wrapper.

This wrapper is very implemented with straightforward shell scripts. No more HPC or Cloud manager is used. We are surveying such tools to enhance SOLVCON-GCE.

**Test Case**
1. Fetch the branch. Tweak the test configuration like this commit https://github.com/tai271828/solvcon-gce/commit/2c4fc9e7c4f225b26ffe1acf4c609c2488da9cb2
2. Source your gce. Issue the command `gcluster 2 -n` to build the cluster which has 1 master node and 2 client nodes.

**Expected Result**
3 GCE instances should be built. They are 1 master node and 2 client nodes respectively. SOLVCON unit test `test_remote.py` should be launched automatically and pass like this:

```
S..S
----------------------------------------------------------------------
Ran 4 tests in 11.872s

OK (SKIP=2)
```

`gcelist` should give there 3 instances:

```
NAME                      ZONE          MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP      STATUS
solvcon-cluster-client-1  asia-east1-c  n1-standard-1               10.140.0.3   104.199.169.104  RUNNING
solvcon-cluster-client-2  asia-east1-c  n1-standard-1               10.140.0.4   104.199.223.193  RUNNING
solvcon-cluster-master    asia-east1-c  n1-standard-1               10.140.0.2   104.155.228.56   RUNNING

```

The two test have been run are test cases `test_dsoln` and `test_dsoln_cluster`. You could also issue `gssh  solvcon-cluster-client-1` to login the instance, `export SOLVCON_NODELIST=~/solvcon-remote/node_list`, and then `nosetests ftests/parallel/test_remote.py -v` to confirm those two test cases were run.

### CODE REVIEW

File: etc/gcerc [hunk 1]
```diff
@@ -124,6 +124,152 @@ echo \"run /var/lib/bootstrap-gce.sh\" ; \
...
+  shift # past argument or value
+  done
+
+  if [ -n "$help" ]; then
+    echo $usage
+    return
+  fi
+
+  script_location="$HOME/opt/gce/bin/admin/bootstrap-cluster.sh"
+  ip_address=`gce-guess-master`
```
Comment (tai271828) [inline etc/gcerc:204]: Where we use the helper script to guess the master IP address instead of pure hard-code.

File: bin/gce-guess-master [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (tai271828) [inline bin/gce-guess-master:1]: A helper script used at here https://github.com/solvcon/solvcon-gce/pull/16/files#r79881209

File: etc/gcerc [hunk 3]
```diff
@@ -124,6 +124,152 @@ echo \"run /var/lib/bootstrap-gce.sh\" ; \
...
+
+  if [ -n "$help" ]; then
+    echo $usage
+    return
+  fi
+
+  echo "Building master node ..."
+  # Do not use underscore for instance naming. GCE kits do not allow underscore
+  # for field conflict.
+  gstart "solvcon-cluster-master"
```
Comment (tai271828) [inline etc/gcerc:253]: The master node name is hard-coded. This could be more flexible. I will revisit it.

File: etc/gcerc [hunk 4]
```diff
@@ -124,6 +124,152 @@ echo \"run /var/lib/bootstrap-gce.sh\" ; \
...
+  gmaster "solvcon-cluster-master"
+  for (( node_index=1; node_index<=$instnumber; node_index++ ))
+  #for node_index in {1..$instnumber};
+  do
+    echo "Building client node $node_index ..."
+    gstart "solvcon-cluster-client-$node_index"
```
Comment (tai271828) [inline etc/gcerc:259]: Ditto. The name could be more flexible.

File: etc/gcerc [hunk 5]
```diff
@@ -124,6 +124,152 @@ echo \"run /var/lib/bootstrap-gce.sh\" ; \
...
+    gclient "solvcon-cluster-client-$node_index"
+  done
+  echo "Complete to build a SOLVCON cluster!"
+  if [ -n "$nose"]; then
+    cmd="source activate ; \
+    export SOLVCON_NODELIST=$HOME/solvcon-remote/node_list ; \
```
Comment (tai271828) [inline etc/gcerc:265]: Ditto. Node list path and name should be more flexible to be specified by uses.

File: etc/gcerc [hunk 6]
```diff
@@ -124,6 +124,152 @@ echo \"run /var/lib/bootstrap-gce.sh\" ; \
...
+    cd $HOME/solvcon-remote/solvcon-src ; \
+    python setup.py build_ext --inplace; \
+    nosetests ftests/parallel/test_remote.py"
```
Comment (tai271828) [inline etc/gcerc:268]: Maybe I should turn on verbose mode for test_remote.py test. Besides, maybe I should include the completed unit tests and ftests at this step.

File: etc/gcerc (hunk 3 above)
Comment (yungyuc) [reply etc/gcerc:253]: Agree we should parameterize the cluster / master name.  Do you know how kubernetes treats this kind of things?

File: etc/gcerc (hunk 6 above)
Comment (yungyuc) [reply etc/gcerc:268]: Sounds good.  We will have a lot of lines to stdout.  Perhaps it's necessary.

File: etc/gcerc [hunk 7]
```diff
@@ -124,6 +124,152 @@ echo \"run /var/lib/bootstrap-gce.sh\" ; \
...
+  gmaster "solvcon-cluster-master"
+  for (( node_index=1; node_index<=$instnumber; node_index++ ))
+  #for node_index in {1..$instnumber};
+  do
+    echo "Building client node $node_index ..."
+    gstart "solvcon-cluster-client-$node_index"
+    gclient "solvcon-cluster-client-$node_index"
+  done
+  echo "Complete to build a SOLVCON cluster!"
+  if [ -n "$nose"]; then
```
Comment (yungyuc) [inline etc/gcerc:263]: Missed a space before `]`

**Review by yungyuc — APPROVED** [review]

**Pull request merged**